lib/rule-context.js

```diff
@@ -82,6 +83,9 @@ function RuleContext(ruleId, eslint, severity, options, settings, parserOptions,
     this.parserPath = parserPath;
     this.meta = meta;
 
+    // create a separate copy and freeze it (it's not nice to freeze other people's objects)
+    this.parserServices = Object.freeze(Object.assign({}, parserServices));
```

Comment:
@platinumazure In your initial review, you asked if it should be frozen, so I changed it to be frozen. After I did that, you asked if a test should check for it to be frozen, and I added that. In the original version, I wasn't freezing or cloning, but it's bad practice to freeze an object you didn't create, so if I changed to freeze then I also needed to clone.